File: 56. Merge Intervals/Solution.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals = sorted(intervals, key=lambda x : x[0])
        result = [intervals[0]]
        for x,y in intervals[1:]:
            if x<=result[-1][1]:
                result[-1][1] = max(result[-1][1], y)
            else:
                result.append([x,y])
        return result

File: 56. Merge Intervals/Solution.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (x,y) -> Integer.compare(x[0], y[0]));
        List<int[]> result = new ArrayList<>();
        result.add(intervals[0]);
        for(int i=1;i<intervals.length;i++){
            int[] temp = result.get(result.size()-1);
            if(intervals[i][0]<=temp[1]){
                temp[1] = Math.max(temp[1], intervals[i][1]);
                result.set(result.size()-1,temp);
            }else{
                result.add(intervals[i]);
            }
        }
        return result.toArray(new int[result.size()][2]);
    }
}
